client/src/http.rs

```diff
@@ -0,0 +1,32 @@
+use std::{borrow::Borrow, collections::HashMap};
+
+use eyre::Result;
+pub use http::{Method, Response, StatusCode};
+
+/// Type alias for HTTP headers hash map
+pub type Headers = HashMap<String, String>;
+
+/// General trait for building http-requests.
+///
+/// To use custom builder with client, you need to implement this trait for some type and pass it
+/// to the client that will fill it.
+pub trait RequestBuilder {
```

Comment:
This is not how people normally construct a builder. 

Firstly, it is preferred to use the chained types and .field(value) syntax. Which would make it a little more obvious which field is which. Moreover, if you break the construction up into stages, you could make the conversions early and make the builder non-generic.

Reply:
Done